confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSource.java

```diff
@@ -79,7 +102,32 @@ public JavaDStream<StructuredRecord> getStream(StreamingContext context) throws
     collector.getOrThrowException();
 
     context.registerLineage(conf.referenceName);
-    return ConfluentStreamingSourceUtil.getStructuredRecordJavaDStream(context, conf, outputSchema, collector);
+    JavaInputDStream<ConsumerRecord<Object, Object>> javaInputDStream = ConfluentStreamingSourceUtil
+      .getConsumerRecordJavaDStream(context, conf, outputSchema, collector, getStateSupplier(context));
+
+    JavaDStream<StructuredRecord> javaDStream;
```

Comment:
I think we can improve the code here to create the javaDStream only once . If the schema registry url is not null, this will create the javaDStream and then re-assign the value.

Reply:
Added the else block to assign `javaDStream` for RecordTransform 2698de880d1d31e02abdfb05d2c3868a321c46c0